storage/gcp/DESIGN.md

```diff
@@ -0,0 +1,71 @@
+# GCP Design
+
+This document describes how the storage implementation for running Tessera on Google Cloud
+is intended to work.
+
+## Overview
+
+This design takes advantage of GCS for long term storage and low cost & complexity serving of read traffic, 
+but leverages something more _transactional_ for coordinating the cluster.
+
+New entries flow in from the binary built with Tessera into transactional storage, where they're held
+temporarily to batch them up, and then assigned sequence numbers en bloc as each batch is flushed.
```

Comment:
consider replacing `en bloc` with something more standard?

Reply:
just removed it, I think it still makes sense without it